### PR TITLE
support global thread pool in onnx

### PR DESCRIPTION
**Pull request by dzhao**
This will expose onnx's global thread pool support. This option great improves the latency when serving multiple models.

### CODE REVIEW

File: src/environment.rs [hunk 1]
```diff
@@ -76,7 +79,44 @@ impl Environment {
...
+		(status, env_ptr)
+	}
+
+	fn create_global_thread_pool_env(name: &str, log_level: LoggingLevel, mut options: HashMap<String, String>) -> (OrtStatusPtr, *mut sys::OrtEnv) {
+		let mut env_ptr: *mut sys::OrtEnv = std::ptr::null_mut();
+		let mut thread_options: *mut sys::OrtThreadingOptions = std::ptr::null_mut();
+		let cname = CString::new(name.clone()).unwrap();
+		let create_thread_options = ortsys![CreateThreadingOptions];
+		let release_thread_options = ortsys![ReleaseThreadingOptions];
+		let create_env_with_global_thread_pool = ortsys![CreateEnvWithGlobalThreadPools];
```
Comment (sudo-carson) [inline src/environment.rs:99]: [`CreateEnvWithCustomLoggerAndGlobalThreadPools`](https://onnxruntime.ai/docs/api/c/struct_ort_api.html#a71336a3b27040097136a6fb40cf056c1)?

File: src/environment.rs [hunk 2]
```diff
@@ -76,7 +79,44 @@ impl Environment {
...
+		let set_global_intra_op_thread_affinity	 = ortsys![SetGlobalIntraOpThreadAffinity];
+		let set_global_intra_op_num_threads = ortsys![SetGlobalIntraOpNumThreads];
+		let set_global_inter_op_num_threads = ortsys![SetGlobalInterOpNumThreads];
+		let set_global_spin_control = ortsys![SetGlobalSpinControl];
+		unsafe { create_thread_options(&mut thread_options); }
+		options.remove("inter_op_parallelism").map(|v| unsafe { set_global_inter_op_num_threads(thread_options, v.parse::<i32>().unwrap()) });
```
Comment (sudo-carson) [inline src/environment.rs:105]: Bit pedantic, but if the keys are known, the options should maybe be a struct

File: src/environment.rs (hunk 1 above)
Comment (dzhao) [reply src/environment.rs:99]: oh sure

Comment (dzhao) [reply src/environment.rs]: ok fixed.. not sure why the pr still shows the old code

File: src/environment.rs (hunk 2 above)
Comment (dzhao) [reply src/environment.rs]: I'll leave you to refactor? :) for me String is good and flexible enough. Or else it becomes a bit verbose.

**Review by sudo-carson — APPROVED** [review]

**Pull request merged**